lib/components/normal-components/Switch.ts

```diff
@@ -0,0 +1,72 @@
+import { switchProps } from "@tscircuit/props"
+import { NormalComponent } from "../base-components/NormalComponent/NormalComponent"
+import type { BaseSymbolName } from "lib/utils/constants"
+import { z } from "zod"
+
+export interface ExtendedSwitchProps extends z.infer<typeof switchProps> {
+  spst?: boolean
+  spdt?: boolean
+  dpst?: boolean
+  dpdt?: boolean
+}
+
+export class Switch extends NormalComponent<typeof switchProps> {
+  get config() {
+    let baseSymbolName: BaseSymbolName
+
+    const { spst, spdt, dpst, dpdt, type } = this.props as ExtendedSwitchProps
+
+    let switchType: "spst" | "spdt" | "dpst" | "dpdt" = "spst"
+
+    if (spst) {
+      switchType = "spst"
+    } else if (spdt) {
+      switchType = "spdt"
+    } else if (dpst) {
+      switchType = "dpst"
+    } else if (dpdt) {
+      switchType = "dpdt"
+    } else if (type) {
+      switchType = type
+    }
+
+    switch (switchType) {
+      case "spst":
+        baseSymbolName = "SPST_switch"
+        break
+      case "spdt":
+        baseSymbolName = "SPDT_switch"
+        break
+      case "dpst":
+        baseSymbolName = "dpst_switch"
+        break
+      case "dpdt":
+        baseSymbolName = "dpdt_switch"
+        break
+      default:
+        baseSymbolName = "SPST_switch"
+        break
+    }
```

Comment:
use _parsedProps, you should be able to reduce this function a lot. If you have to, create `_getSwitchType` as a util method. Also use this pattern
```tsx
const baseSymbolName = {
  spst: "SPST_switch",
  spdt: "SPDT_switch",
  // ...
}[switchType] ?? "SPST_switch"
```

Also **the symbol names should NOT have capitals, make sure this gets refactored at some point!**